sa/storage-authority.go

```diff
@@ -521,25 +565,24 @@ func (ssa *SQLStorageAuthority) AddCertificate(certDER []byte) (digest string, e
 		return
 	}
 	serial := fmt.Sprintf("%032x", parsedCertificate.SerialNumber)
+	digest = core.Fingerprint256(certDER)
 
-	tx, err := ssa.db.Begin()
+	cert := &Certificate{serial, digest, certDER, time.Now()}
```

Comment:
This and the CertificateStatus declaration below are long enough that they should be exploded onto multiple lines, with field names.